src/enable_variable_ops/ngraph_rewrite_pass.cc

```diff
@@ -249,17 +249,39 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
       return Status::OK();
     }
 
+    // Get backend + its configurations
+    // to be attached to the nodes
+    std::unordered_map<std::string, std::string> config_map;
+    string backend_name = BackendManager::GetCurrentlySetBackendName();
+    const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+    if (ng_backend_env_value != nullptr) {
+      string backend_env = std::string(ng_backend_env_value);
+      if (backend_env.empty() ||
+          !BackendManager::IsSupportedBackend(backend_env)) {
+        return errors::Internal("NGRAPH_TF_BACKEND: ", backend_env,
+                                " is not supported");
+      }
+      backend_name = backend_env;
+    }
+    config_map = BackendManager::GetBackendAttributes(
+        backend_name);  // SplitBackendConfig
+    backend_name = config_map.at("ngraph_backend");
+    config_map.erase("ngraph_backend");
```

Comment:
Why do we `erase` the attribute?

Reply:
backend_name is required by mark for clustering. We use the use backend_name to assign_clusters (ops marked with different backends are not clustered together).  

The other attributes are added directly in encapsulate clusters through the config_map.

Reply:
Oh, I see. `config_map` in `EncapsulateClusters` is not expected to contain `ngraph_backend`